assets-sync.js

```diff
@@ -709,7 +709,7 @@
         }
 
         try {
-            const nameParts = asset.name.split("/");
+            const nameParts = asset.name.split(/[\/\\]/);
```

Comment:
Urghhh... `\` shouldn't be in the asset name at all. And we make sure server side to never allow `\` as a part of any filename.

I really should do this todo task I've had since june 2022 😭 
![image](https://github.com/ForgeVTT/fvtt-module-forge-vtt/assets/27990/68ca4d81-42fb-430a-a0ee-5059bbd8c3dc)
I wrote the script for it, and ran it for the specific user who had the issue, but didn't run it for all users... been hesitating because I worry about unintended consequences... 

Just need to run this script... 
https://github.com/ForgeVTT/theforge/blob/master/tools/sanitize-assets-library.js

Reply:
I can understand the concern (though I suspect it would be safe, considering that we have handled a few cases like this for now).

Given that we have some existing cases with `\` from before we controlled those cases more strictly, I think there is still some benefit to be gained from being able to handle the edge cases on the side of the Asset Sync. 

More likely to cause issues, though, is the early exit from `createDirectory`, which I believe can cause the directory to bnot be fully created (resulting in `ENOENT`) even if no `\` exists in the path.

Reply:
Removed early assignment. The issue that this PR addresses could happen regardless of whether back- or forwardslashes are used. Backslashes to be handled separately in script